src/libutil/file-system.hh

```diff
@@ -84,6 +84,7 @@ bool isDirOrInDir(std::string_view path, std::string_view dir);
  */
 struct stat stat(const Path & path);
 struct stat lstat(const Path & path);
+std::optional<struct stat> maybeLstat(const Path & path);
```

Comment:
```suggestion
/**
 * `lstat` the given path if it exists.
 * @return std::nullopt if the path doesn't exist, or an optional containing the result of `lstat` otherwise
 */
std::optional<struct stat> maybeLstat(const Path & path);
```

(not sure about the exact wording, but we should document why it can return an empty optional)